### PR TITLE
Add Discovery Chain Post

### PR DESCRIPTION
**Pull request by larrytamnjong**

### CODE REVIEW

**Comment by larrytamnjong** [comment]
@marcin-krystianc

File: Consul/DiscoveryChain.cs [hunk 1]
```diff
@@ -124,6 +124,36 @@ public class DiscoveryTarget
         public string Name { get; set; }
     }
 
+    public class DiscoveryChainOptions
+    {
+        public string EvaluateInDatacenter { get; set; }
```
Comment (marcin-krystianc) [inline Consul/DiscoveryChain.cs:129]: I think this should not be a part of this structure as it is being used as `[compile-dc](https://developer.hashicorp.com/consul/api-docs/discovery-chain#compile-dc)` which is a query-only parameter.

File: Consul/DiscoveryChain.cs [hunk 2]
```diff
@@ -159,6 +189,37 @@ public Task<QueryResult<DiscoveryChainResponse>> Get(string name, CancellationTo
...
+        /// <param name="options">Discovery Chain Options</param>
+        /// <param name="q">Write Options</param>
+        /// <param name="ct">Cancellation Token</param>
+        /// <returns>An empty write result</returns>
+        public Task<WriteResult<DiscoveryChainResponse>> Get(string name, DiscoveryChainOptions options, WriteOptions q, CancellationToken ct = default)
+        {
+            var request = _client.Post<DiscoveryChainOptions, DiscoveryChainResponse>($"/v1/discovery-chain/{name}", options, q);
+            if (options != null && options.EvaluateInDatacenter != null)
+            {
+                request.Params["compile-dc"] = options.EvaluateInDatacenter;
```
Comment (marcin-krystianc) [inline Consul/DiscoveryChain.cs:218]: This should be a separate argument passed to the `Get` method.

**Review by marcin-krystianc — COMMENTED** [review]
Hi @larrytamnjong,

I left comments about `compile-dc` argument, looks good otherwise!

**Comment by larrytamnjong** [comment]
> Hi @larrytamnjong,
> 
> I left comments about `compile-dc` argument, looks good otherwise!

Thanks @marcin-krystianc I have made an update based on your feedback.

File: Consul.Test/DiscoveryChainTest.cs [hunk 3]
```diff
@@ -99,6 +99,144 @@ public async Task DiscoveryChain_Get()
...
+            targetChain = chain.Targets["web." + defaultPart + ".dc2"];
+            Assert.Equal(targetCheck.ID, targetChain.ID);
+            Assert.Equal(targetCheck.Service, targetChain.Service);
+            Assert.Equal(targetCheck.Namespace, targetChain.Namespace);
+            Assert.Equal(targetCheck.Datacenter, targetChain.Datacenter);
+            Assert.Equal(targetCheck.SNI, targetChain.SNI);
+            Assert.Equal(targetCheck.Name, targetChain.Name);
         }
     }
+
```
Comment (marcin-krystianc) [inline Consul.Test/DiscoveryChainTest.cs:239]: Nit: Please remove extra empty lines

File: Consul.Test/DiscoveryChainTest.cs [hunk 4]
```diff
@@ -99,6 +99,144 @@ public async Task DiscoveryChain_Get()
...
 }
+
```
Comment (marcin-krystianc) [inline Consul.Test/DiscoveryChainTest.cs:241]: Nit: Please remove extra empty lines

File: Consul.Test/DiscoveryChainTest.cs [hunk 5]
```diff
@@ -99,6 +99,144 @@ public async Task DiscoveryChain_Get()
...
+
```
Comment (marcin-krystianc) [inline Consul.Test/DiscoveryChainTest.cs:242]: Nit: Please remove extra empty lines

**Review by marcin-krystianc — COMMENTED** [review]
Looks good, please just remove those unnecessary empty lines.

**Review by marcin-krystianc — APPROVED** [review]
Nice 👍

**Pull request merged**